interop-testing/src/test/java/io/grpc/testing/integration/MoreInProcessTest.java

```diff
@@ -247,7 +247,7 @@ public void onCompleted() {
         .onNext(StreamingInputCallRequest.getDefaultInstance());
 
     assertTrue(finishLatch.await(900, TimeUnit.MILLISECONDS));
-    assertEquals(Status.UNKNOWN, Status.fromThrowable(throwableRef.get()));
+    assertEquals(Status.CANCELLED.getCode(), Status.fromThrowable(throwableRef.get()).getCode());
```

Comment:
Right now calling `serverStream.cancel()` will propagate a [CANCEL HTTP/2 error](https://github.com/grpc/grpc-java/blob/35665af/netty/src/main/java/io/grpc/netty/NettyServerHandler.java#L658) to the remote. It'd probably be better if it was [INTERNAL_ERROR](http://httpwg.org/specs/rfc7540.html#ErrorCodes), which will get [mapped to a INTERNAL](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#errors) grpc status code. That way the client would get a signal that it shouldn't retry.

I think it'd be fair to map gRPC's DEADLINE_EXCEEDED and CANCELLED error codes to HTTP/2's CANCEL, and everything else map to HTTP/2's INTERNAL_ERROR. Right now we use [DEADLINE_EXCEEDED for when deadline expires](https://github.com/grpc/grpc-java/blob/5d5e8c6/core/src/main/java/io/grpc/internal/ServerImpl.java#L496). That mapping could be in `NettyServerHandler.cancelStream()`.